Final mile procedures and updates

victoriaatraft opened the pull request:
## Summary of Changes
New PR to correct for some unwanted conflicts with develop 
Closes #2801

victoriaatraft committed:
Final mile procedures and updates

additional updates to schemas and friendly names

victoriaatraft committed:
Updating validators

Adding back validators that got removed on previous commit

elipe17 committed:
- update incorrect name

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

elipe17 committed:
Merge branch 'friendlynamefixes2' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

elipe17 committed:
- Fix merge conflicts and update tests

elipe17 committed:
- fix lint

reitermb reviewed (approved):
Looks great to me! Thanks Victoria! And thank you Eric for the test fixes! 

Over to @ADPennington for official QASP approval

ADPennington committed:
Merge branch 'develop' into friendlynamefixes2

ADPennington committed:
Merge branch 'develop' into friendlynamefixes2

ADPennington committed:
Merge branch 'develop' into friendlynamefixes2

ADPennington committed:
Merge branch 'develop' into friendlynamefixes2

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m2.py:
```diff
@@ -184,21 +183,18 @@
                     startIndex=21,
                     endIndex=29,
                     required=True,
-                    validators=[validators.intHasLength(8),
```
@victoriaatraft @elipe17 why did we change this validator?

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m3.py:
```diff
@@ -139,22 +139,21 @@
-                        validators.dateMonthIsValid(),
-                        validators.dateDayIsValid()
-                        ]
+            validators=[
```
@elipe17 @victoriaatraft why this change?

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m3.py:
```diff
@@ -453,22 +453,21 @@
-                        validators.dateMonthIsValid(),
-                        validators.dateDayIsValid()
-                        ]
+            validators=[
```
@elipe17 @victoriaatraft same question as above.

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m5.py:
```diff
@@ -152,22 +152,21 @@
-                                validators.dateMonthIsValid(),
-                                validators.dateDayIsValid()
-                                ],
+                    validators=[
```
@elipe17 @victoriaatraft same questions as above.

victoriaatraft replied:
@ADPennington I will follow up with @EricYagerPersonalRepo on these right now. I know we had some conflicts that he helped us with.

ADPennington commented:
adding blocked tag pending feedback on validator changes.

victoriaatraft commented:
> adding blocked tag pending feedback on validator changes.

okay. I 100% deleted them, so sorry about that. @elipe17 will graciously add them back in so it can be good to go.

elipe17 committed:
- Re-added validators

ADPennington commented:
@victoriaatraft a few of these friendly names are still quite verbose. lets discuss options for a path forward at next UX sync. cc: @lfrohlich

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t4.py:
```diff
@@ -150,8 +150,8 @@
                     item="14",
-                    name="BLANK",
-                    friendly_name="blank",
+                    name="FAMILY_AFFILIATION",
```
this change needs to be prioritized for merging.

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t5.py:
```diff
@@ -392,7 +392,7 @@
                     item="29",
                     name="AMOUNT_EARNED_INCOME",
-                    friendly_name="amount earnedof income",
+                    friendly_name="Amount of Earned Income",
```
prioritize this change for merging

ADPennington reviewed (changes requested):
@elipe17 @victoriaatraft i left comments on a couple of line changes that can be prioritized for  merging, perhaps on another branch. defer to dev on how best to go about that.

victoriaatraft commented:
> @elipe17 @victoriaatraft i left comments on a couple of line changes that can be prioritized for merging, perhaps on another branch. defer to dev on how best to go about that.

Thanks @ADPennington! checked in with @elipe17 and will be tackling this, this afternoon and wrap up by tomorrow morning

elipe17 committed:
- remvoe trailing colon

elipe17 commented:
> @elipe17 @victoriaatraft i left comments on a couple of line changes that can be prioritized for merging, perhaps on another branch. defer to dev on how best to go about that.

@ADPennington, @victoriaatraft I pulled the changes into another branch: `merge-first-changes` and made that branch this branch's base branch. I tagged everyone on the new PR.

victoriaatraft committed:
Updates to the friendly names

added code no longer in use where indicated and text edits

victoriaatraft committed:
Update t2.py

address linting errors to make less than 120 characters

victoriaatraft committed:
t2.py linting fixes

addressing character count

elipe17 committed:
making a dummy change

elipe17 committed:
Merge branch 'friendlynamefixes2' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

elipe17 committed:
Revert "making a dummy change"

This reverts commit 7c6e05581265cec97397ea7b96af529d21759f61.

victoriaatraft committed:
Friendly Name Updates

Shortened friendly names, removed "code not in use", removed "race/ethnicity", shortened benefits being received

victoriaatraft committed:
rewrote 54 job search

victoriaatraft committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

victoriaatraft committed:
finished resolving conflicts

victoriaatraft committed:
updated test_api.py

finding lint issues

victoriaatraft committed:
corrected message

victoriaatraft committed:
updated friendly names

victoriaatraft committed:
Merge branch 'develop' into friendlynamefixes2

elipe17 committed:
- getting test case back its expected state

elipe17 committed:
Merge branch 'develop' into friendlynamefixes2

victoriaatraft reviewed (commented):
Eric pulled the changes into another branch: merge-first-changes and made that branch this branch's base branch and tagged everyone on the new PR. -- ignore, tried to dismiss a request change below

victoriaatraft commented:
@ADPennington @lfrohlich  I know it says change requested, I tried to resolve that but those changes were addressed by Eric :)

ADPennington commented:
> @ADPennington @lfrohlich I know it says change requested, I tried to resolve that but those changes were addressed by Eric :)

thank you.  just one quick question on this. Any suggestions on an efficient way to test this? For context, all of the friendly names have changed with this PR which currently only appear in the error reports. I usually confirm these sorts of changes by submitting a test file that would yield the relevant error messages that have changed. I'm not sure if we have some test files relevant to this PR to help make this a little easier. @victoriaatraft @reitermb @elipe17

elipe17 commented:
> > @ADPennington @lfrohlich I know it says change requested, I tried to resolve that but those changes were addressed by Eric :)
> 
> thank you. just one quick question on this. Any suggestions on an efficient way to test this? For context, all of the friendly names have changed with this PR which currently only appear in the error reports. I usually confirm these sorts of changes by submitting a test file that would yield the relevant error messages that have changed. I'm not sure if we have some test files relevant to this PR to help make this a little easier. @victoriaatraft @reitermb @elipe17

@ADPennington there is not an easy way for us to do this at the moment. Best we can do is submit files that have lots of errors, e.g. `ADS.E2J.FTP1.TS06`, and do a manual verification. The only way we might be able to actually test this programmatically is if we built a file generator that allowed us to specify if we want to fuzz the values for the record types. Then we could generate files that have an error for every single field in each record type quite easily.

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

reitermb committed:
Merge branch 'develop' into friendlynamefixes2

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

ADPennington commented:
unit tests now failing on this @elipe17 @victoriaatraft

elipe17 commented:
> Details

Ah yes, these failures are result of the recent PRs being merged. @ADPennington, @victoriaatraft I will take a look shortly.

elipe17 committed:
- Updated tests to use new friendly names

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t2.py:
```diff
@@ -795,7 +800,8 @@
                     name="SCHOOL_ATTENDENCE_EA",
-                    friendly_name="school attendance - excused absence",
+                    friendly_name="Satisfactory School Attendance: " +
+                    "Hours of Ecused Absences",
```
```suggestion
                    "Hours of Excused Absences",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t2.py:
```diff
@@ -511,6 +511,7 @@
                                 "16",
                                 "17",
                                 "18",
+                                "19",
```
@elipe17 why this change? 

![WPS_49](https://github.com/raft-tech/TANF-app/assets/63075587/b497513f-e61c-416a-8283-30d2e865fd44)

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t3.py:
```diff
@@ -169,7 +169,7 @@
             item="70A",
             name="RACE_HISPANIC",
-            friendly_name="race hispanic",
+            friendly_name="Ethnicity/Race: Hispanic or Latino",
```
@victoriaatraft here you add "ethnicity/race:" but not for M2, M3, T2,  Tribal T2, Tribal T3, M5, T5, and Tribal T5. Please inform what the expected language should be. cc: @elipe17

victoriaatraft replied:
Thank you. I'll updated to remove "ethnicity/race" @ADPennington

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t5.py:
```diff
@@ -182,7 +182,7 @@
                     item="17A",
                     name="RACE_HISPANIC",
-                    friendly_name="race hispanic",
+                    friendly_name="Hisapic or Latino",
```
```suggestion
                    friendly_name="Hispanic or Latino",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m6.py:
```diff
@@ -276,7 +275,7 @@
             item="7B",
             name='NUM_RECIPIENTS',
-            friendly_name='number of recipients',
+            friendly_name='SSP-MOERecipients',
```
```suggestion
            friendly_name='SSP-MOE Recipients',
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t6.py:
```diff
@@ -107,7 +107,7 @@
             item="7A",
             name="ASSISTANCE",
-            friendly_name="total amount of assistance",
+            friendly_name="Total Number of Cash Assistance",
```
```suggestion
            friendly_name="Total Amount of Cash Assistance",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t6.py:
```diff
@@ -517,7 +517,7 @@
             item="5C",
             name="NUM_APPROVED",
-            friendly_name="total number of approved applications",
+            friendly_name="Total Number of Applications",
```
```suggestion
            friendly_name="Total Number of Approved Applications",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t6.py:
```diff
@@ -537,7 +537,7 @@
             item="7C",
             name="ASSISTANCE",
-            friendly_name="total amount of assistance",
+            friendly_name="Total Number of Cash Assistance",
```
```suggestion
            friendly_name="Total Amount of Cash Assistance",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t6.py:
```diff
@@ -125,7 +125,7 @@
             item="10A",
             name="NUM_1_PARENTS",
-            friendly_name="total number of one-parent families",
+            friendly_name="Total Number of No-Parent Families",
```
```suggestion
            friendly_name="Total Number of One-Parent Families",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t6.py:
```diff
@@ -135,7 +135,7 @@
             item="11A",
             name="NUM_NO_PARENTS",
-            friendly_name="total number of no parent families",
+            friendly_name="Total Number of No Parent Families",
```
```suggestion
            friendly_name="Total Number of No-Parent Families",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t6.py:
```diff
@@ -268,7 +268,7 @@
             item="4B",
             name="NUM_APPLICATIONS",
-            friendly_name="total number of applications",
+            friendly_name="Total Number of Applicants",
```
```suggestion
            friendly_name="Total Number of Applications",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t6.py:
```diff
@@ -541,7 +541,7 @@
             item="11C",
             name="NUM_NO_PARENTS",
-            friendly_name="total number of no parent families",
+            friendly_name="Total Number of No parent Families",
```
```suggestion
            friendly_name="Total Number of No-parent Families",
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m7.py:
```diff
@@ -71,7 +71,7 @@
                     item="3",
                     name="TDRS_SECTION_IND",
-                    friendly_name="tdrs section indicator",
+                    friendly_name="SDR Section Indicator",
```
```suggestion
                    friendly_name="TDR Section Indicator",
```

ADPennington commented on tdrs-backend/tdpservice/settings/common.py:
```diff
@@ -194,7 +194,7 @@ class Common(Configuration):
     # Logging
     # set level as 'INFO' if env var is not set
-    LOGGING_LEVEL = os.getenv('LOGGING_LEVEL', 'INFO')
+    LOGGING_LEVEL = 'WARNING'#os.getenv('LOGGING_LEVEL', 'INFO')
```
@elipe17 can the syntax that is commented out be removed?

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py:
```diff
@@ -121,6 +121,8 @@
                         "CASH_AMOUNT",
                         "CC_AMOUNT",
                         "TRANSP_AMOUNT",
+                        "TRANSITION_SERVICES_AMOUNT",
```
```suggestion
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py:
```diff
@@ -121,6 +121,8 @@
                         "CC_AMOUNT",
                         "TRANSP_AMOUNT",
+                        "TRANSITION_SERVICES_AMOUNT",
+                        "OTHER_AMOUNT",
```
```suggestion
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py:
```diff
@@ -197,31 +199,31 @@
                     required=True,
                     validators=[
-                        validators.isInLimits(1, 2),
+                        validators.isInLimits(1, 3),
```
```suggestion
                        validators.isInLimits(1, 2),
```

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py:
```diff
@@ -197,31 +199,31 @@
                     required=True,
                     validators=[
-                        validators.matches(1)
+                        validators.oneOf([1, 2]),
```
please revert. @elipe17

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py:
```diff
@@ -257,19 +259,19 @@
                     required=True,
                     validators=[
-                        validators.isInLimits(1, 2),
+                        validators.isInLimits(1, 3),
```
@elipe17 please revert.

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t2.py:
```diff
@@ -690,7 +689,7 @@
                     item="65B",
                     name="UNEARNED_SOCIAL_SECURITY",
-                    friendly_name="unearned social security",
+                    friendly_name="Amount of Unearned Income: SSI",
```
```suggestion
                    friendly_name="Amount of Unearned Income: Social Security",
```

ADPennington reviewed (changes requested):
@victoriaatraft @elipe17 changes requested

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/ssp/m6.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tanf/t2.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
- remove 19

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tanf/t6.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tanf/t6.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tanf/t6.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t6.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t6.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t6.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t6.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
- revert debug comment

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
- revert validator changes

elipe17 committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t2.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

elipe17 committed:
Merge branch 'friendlynamefixes2' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

elipe17 commented:
@victoriaatraft and @ADPennington I resolved the suggestions/changes that were in my purview. I left a few open for Victoria/@reitermb to address.

elipe17 committed:
- Fixing unresolved merge conflict

elipe17 committed:
- resolved merge conflicts causing failure

elipe17 committed:
- resolve merge conflict causing failure

victoriaatraft committed:
Update tdrs-backend/tdpservice/parsers/schema_defs/tanf/t5.py

Co-authored-by: Alex P.  <63075587+ADPennington@users.noreply.github.com>

victoriaatraft committed:
Update t3.py

Removed Ethnicity/Race

victoriaatraft commented:
> @elipe17 @ADPennington resolved open suggestion and made the last to Alex's open comment. We're removing "Race/Ethnicity" thank you

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

elipe17 committed:
- resolved merge conflicts

elipe17 committed:
Merge branch 'friendlynamefixes2' of https://github.com/raft-tech/TANF-app into friendlynamefixes2

elipe17 committed:
- fixed failing tests

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m3.py:
```diff
@@ -252,7 +253,8 @@
             item="65B",
             name='RECEIVE_SSI',
-            friendly_name="receives SSI",
+            friendly_name="Receives Disability Benefits: SSI Under Title XVI-SSI or " +
```
@victoriaatraft @elipe17 please confirm if this change is the correct one or the one located in [line 574](https://github.com/raft-tech/TANF-app/blob/880f13b5032f0e178ea45c1e9831279654434fcb/tdrs-backend/tdpservice/parsers/schema_defs/ssp/m3.py#L574)

victoriaatraft committed:
Updated mp3.py

Shortened Receives Disability Benefits: SSI or AABD

elipe17 committed:
- fixed missed comma error

elipe17 committed:
Merge branch 'develop' into friendlynamefixes2

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m6.py:
```diff
@@ -174,7 +174,6 @@
 s2 = RowSchema(
     record_type="M6",
     document=SSP_M6DataSubmissionDocument(),
-    quiet_preparser_errors=True,
```
@elipe17 why this change?

elipe17 replied:
Looks like this got clobbered in a merge somehow.

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/ssp/m6.py:
```diff
@@ -339,7 +338,6 @@
 s3 = RowSchema(
     record_type="M6",
     document=SSP_M6DataSubmissionDocument(),
-    quiet_preparser_errors=True,
```
@elipe17 same question

elipe17 replied:
Looks like this got clobbered in a merge somehow.

ADPennington commented on tdrs-backend/tdpservice/parsers/schema_defs/tanf/t2.py:
```diff
@@ -494,11 +494,11 @@
                     type="string",
                     startIndex=68,
                     endIndex=70,
-                    required=True,
```
this field is not optional. @elipe17

ADPennington reviewed (changes requested):
@elipe17 a few changes requested. Just want to ensure that we are only making friendly name changes here. when these changes are fixed. please merge. @cc: @victoriaatraft @robgendron

elipe17 committed:
- fixed issues

ADPennington reviewed (approved):
🚀
